crates/rspack_plugin_dev_friendly_split_chunks/src/plugin.rs

```diff
@@ -64,7 +64,65 @@ impl Plugin for DevFriendlySplitChunksPlugin {
     });
 
     // The number doesn't go through deep consideration.
-    const MAX_MODULES_PER_CHUNK: usize = 3000;
+    const MAX_MODULES_PER_CHUNK: usize = 500;
+    // About 5mb
+    const MAX_SIZE_PER_CHUNK: f64 = 5000000.0;
+
+    // First we group modules by MAX_MODULES_PER_CHUNK
+
+    let split_modules = shared_modules
+      .par_chunks(MAX_MODULES_PER_CHUNK)
+      .flat_map(|modules| {
+        let chunk_size: f64 = modules
+          .iter()
+          .map(|m| {
+            let module = compilation
+              .module_graph
+              .module_by_identifier(&m.module)
+              .expect("Should have a module here");
+
+            // Some code after transpiling will increase it's size a lot.
+            let coefficient = match module.module_type() {
+              // 5.0 is a number in practice
+              rspack_core::ModuleType::Jsx => 5.0,
```

Comment:
Maybe we could figure out a better way for this. I will take this as a workaround.